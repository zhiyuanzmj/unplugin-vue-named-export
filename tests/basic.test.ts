import path from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupRemoveVueFilePathPlugin,
  RollupVue,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import VueReactivityFunction from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    ['tests/fixtures/*.vue'],
    (args, id) =>
      rollupBuild(id, [
        RollupVue(),
        VueReactivityFunction(),
        RollupRemoveVueFilePathPlugin(),
        RollupEsbuildPlugin({
          target: 'esnext',
        }),
      ]),
    { cwd: path.resolve(__dirname, '..'), promise: true }
  )
})
