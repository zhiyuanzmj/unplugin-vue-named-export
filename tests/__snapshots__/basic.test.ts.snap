// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"import { defineComponent, ref, watch, openBlock, createElementBlock, toDisplayString, unref } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

"use strict";
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "basic",
  setup(__props) {
    function useApi(defaultName = ref("")) {
      const id2 = ref(1);
      const name2 = ref(defaultName);
      return {
        id: id2,
        name: name2
      };
    }
    const { id, name } = $(useApi());
    id === 1;
    useApi($$(name));
    watch($$(name), () => {
    });
    defineExpose({
      name: $$(name)
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "div",
        null,
        toDisplayString(unref(name)),
        1
        /* TEXT */
      );
    };
  }
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;
